src/resizable_array.jl

```diff
@@ -181,14 +181,17 @@ function Base.iterate(array::ResizableArray, state)
         return nothing
     end
     nindex, nstate = niterate
+    while !isassigned(array, nindex.I...)
```

Comment:
This is very dangerous, lets discuss tomorrow ?, I didn't do this operation for a reason. Otherwise the `zip` with a `Matrix` is error prone. I wanted it to throw an error if something is not _assigned_. Maybe we should make a special method for map? Or we could add an option to the iterate method to skip things that aren't set, and then map could use that.

Reply:
E.g. this branch:
```
julia> s = GraphPPL.ResizableArray(Float64, Val(2))
0×0 GraphPPL.ResizableArray{Float64, Vector{Vector{Float64}}, 2}

julia> s[1, 1] = 1.0;
julia> s[2, 2] = 2.0;
julia> s[3, 3] = 3.0;

julia> map(sum, zip(s, Diagonal(ones(3))))
3×3 Matrix{Float64}:
 2.0           2.0  0.0
 2.62258e-314  1.0  0.0
 2.62258e-314  3.0  2.2241e-314
```
On the main it throws (which is better than silently getting wrong result):
```julia
julia> map(sum, zip(s, Diagonal(ones(3))))
ERROR: BoundsError: attempt to access 1-element Vector{Float64} at index [2]
```

The `zip` operation is used in `ReactiveMP` to connect `ResizableArray` with some data. So a user might give `Diagonal` data and get some wrong results instead of an error